pages/api/registry/index.ts

```diff
@@ -0,0 +1,108 @@
+
+  const allowedTokens = await registry.getAllowedTokens()
+  
+  if(allowedTokens) {
+    const addTransactionalTokens = []
+    const removeTransactionalTokens = []
+    const removeRewardTokens = []
+    const addRewardTokens = []
+
+    if(allowedTokens?.transactional?.length > 0) await allowedTokens?.transactional?.map((address) => {
+      const valid = tokens?.find(token => token.address === address && token.isTransactional === true)
+      if(!valid) addTransactionalTokens.push(address)
+    })
+
+    if(allowedTokens?.reward?.length > 0) await allowedTokens?.reward?.map((address) => {
+      const valid = tokens?.find(token => token.address === address && token.isTransactional === false)
+      if(!valid) addRewardTokens.push(address)
+    })
+
+    if(addTransactionalTokens.length > 0){
+      for (const address of addTransactionalTokens) {
+        const token = await DAOService.getERC20TokenData(address);
+        await Database.tokens.create({...token, isTransactional: true})
+      }
+    }
+
+    if(addRewardTokens.length > 0){
+      for (const address of addRewardTokens) {
+        const token = await DAOService.getERC20TokenData(address);
+        await Database.tokens.create({...token, isTransactional: false})
+      }
+    }
+
+    const newTokens = await Database.tokens.findAll();
+
+    if(!newTokens) return res.status(500).json("Error to get new tokens from the database")
+
+    await newTokens?.map(({address}) => {
+      const validTransactional = allowedTokens?.transactional?.find(tAddress => tAddress === address)
+      if(!validTransactional) removeTransactionalTokens.push(address)
+  
+      const validReward = allowedTokens?.reward?.find(rAddress => rAddress === address)
+      if(!validReward) removeRewardTokens.push(address)
+    })
+
+    if(removeTransactionalTokens.length > 0) {
+      for (const address of removeTransactionalTokens) {
+        const token = await Database.tokens.findOne({
+            where: {
+              address: address,
+              isTransactional: true,
+            }
+        });
+        if (token) await Database.networkTokens.destroy({ where : { tokenId: token.id }})
+        .catch(() => console.log('Error synchronizing token deletion'))
+      }
+    }
+    if(removeRewardTokens.length > 0) {
+      for (const address of removeRewardTokens) {
+        const token = await Database.tokens.findOne({
+              where: {
+                address: address,
+                isTransactional: false,
+              }
+        });
+        if (token) await Database.networkTokens.destroy({ where : { tokenId: token.id }})
+        .catch(() => console.log('Error synchronizing token deletion'))
+      }
+    }
+  }
+
+
+  return res.status(200).json("Update allowed tokens in database");
+}
+
+async function UpdateAllowedTokens(req: NextApiRequest,
+                                   res: NextApiResponse) {
+  switch (req.method.toLowerCase()) {
```

Comment:
You're missing a POST method to _change_ the database. GET should only return, POST can change.

You also need to make sure that the person who's making this change to the database actually _can_ make this change - otherwise _anyone_ would be able to make this change.

Reply:
@MarcusviniciusLsantos like we spoke, this script belongs on the webnetwork-events